HumanCellAtlas/metadata-schema
# Adjust schemas based on feedback, manually roll back project

This PR addresses comments and concerns raised in the review of #1371 

Changes are:
* Manually removed all traces of the `data_use_restrictions` field and `data_use_ontology module` from:
    * changelog
    * json browser
    * property migrations 
* Manually wound back the project type schema to 14.2.0 in `versions.json`
* Implemented suggested wording changes to:
    * `file_core.file_source.description`
    * `matrix.description`
    * `matrix.derivation_process.description`
    * `analysis_file.matrix_cell_count.guidelines` & `analysis_file.matrix_cell_count.description`
    * `project.ega_accessions.description`
 
 * added to the enum for the `file_core.file_source`

* For `src/human_readable_json.py` script:
   - L120-125: Added enum_values variable to account for the different scenarios an enum could be present
   - L133: Changed the oneliner for the variable

This last update was tested here https://github.com/HumanCellAtlas/metadata-schema/pull/1379 but we decided to push everything as one PR

I did not fill out the update log as I wasn't sure if we want to increment the patch version for these changes or leave them as is, i.e. we determine that our initial release to staging is a 'soft' release that we can iterate on and update based on feedback before making a 'final' release on the same versions? (If we decide this is ok we should probably add it to the spec)

After this PR is reviewed I believe we will need to delete the following released staging schemas from https://schema.staging.data.humancellatlas.org/:
* 2021-03-16T14:35:35.000Z        5.8 kB         type/project/14.2.0/project
* 2021-03-24T11:09:02.000Z        6.1 kB         type/project/14.3.0/project
* 2021-03-25T15:03:15.000Z        5.8 kB         type/project/15.0.0/project
* 2021-03-24T11:08:54.000Z        2.0 kB         module/ontology/1.0.0/data_use_ontology
* 2021-03-24T13:35:32.000Z        2.7 kB         module/protocol/1.0.0/matrix
* 2021-03-24T13:35:34.000Z        1.8 kB         type/file/6.3.0/analysis_file
* 2021-03-24T13:35:23.000Z        2.1 kB         core/file/6.2.0/file_core

We would then need to re-trigger the publisher so the updated versions of these schemas are published. (apart from the data_use_ontology which has been deleted).

### Test data changes required
* ensure the test data refers to the version 14.2.0 project type schema.
* Ensure test data has correct enum values from the revised `file_core.file_source` enum

Then we would be ready to merge and for request re-review. 

I would suggest squash and merge to minimise commits.
